topostats/run_topostats.py

```diff
@@ -303,186 +172,170 @@ def process_scan(
+                            full_image_data=grains.images["gaussian_filtered"].T,
+                            grains=grains.directions[direction]["labelled_regions_02"],
+                            filename=filtered_image.filename,
+                            pixel_size=filtered_image.pixel_to_nm_scaling,
+                            **dnatracing_config,
+                        )
+                        dna_traces[direction].trace_dna()
+                        tracing_stats[direction] = traceStats(trace_object=dna_traces[direction], image_path=image_path)
+                        tracing_stats[direction].df["threshold"] = direction
+                        # tracing_stats[direction].save_trace_stats(_output_dir / filtered_image.filename / direction)
+
+                    if "lower" in grainstats.keys():
+                        tracing_stats_df = pd.concat([tracing_stats["lower"].df, tracing_stats["upper"].df])
+                    else:
+                        tracing_stats_df = tracing_stats["upper"].df
+                    LOGGER.info(
+                        f"[{filtered_image.filename}] : Combining {direction} grain statistics and dnatracing statistics"
+                    )
+                    results = grainstats_df.merge(tracing_stats_df, on=["Molecule Number", "threshold"])
+                else:
+                    results = grainstats_df
+                    results["Image Name"] = filtered_image.filename
+                    results["Basename"] = image_path.parent
+
+            except Exception:
+                # If no results we need a dummy dataframe to return.
                 LOGGER.info(
-                    f"[{filtered_image.filename}] : Combining {direction} grain statistics and dnatracing statistics"
+                    f"[{filtered_image.filename}] : Errors occurred attempting to calculate grain statistics and DNA tracing statistics."
                 )
-                results = grainstats_df.merge(tracing_stats_df, on=["Molecule Number", "threshold"])
-            else:
-                results = grainstats_df
-                results["Image Name"] = filtered_image.filename
-                results["Basename"] = image_path.parent
-                # results.to_csv(_output_dir / filtered_image.filename / direction / "all_statistics.csv")
-                # LOGGER.info(
-                #    f"[{filtered_image.filename}] : Combined statistics saved to {str(_output_dir)}/{filtered_image.filename}/{direction}/all_statistics.csv"
-                # )
-
-        except Exception:
-            # If no results we need a dummy dataframe to return.
-            LOGGER.info(
-                f"[{filtered_image.filename}] : Errors occurred attempting to calculate grain statistics and DNA tracing statistics."
-            )
-            results = create_empty_dataframe()
+                results = create_empty_dataframe()
 
     # Optionally plot all stages
-    if save_plots:
+    if plotting_config["run"]:
```

Comment:
Still have `save_plots` at the top of the config file although `save` have been replaced by `run`. Is this necessary?